### PR TITLE
Add instant payment method for iOS

### PR DESCRIPTION
**Pull request by Robert-SD**

### CODE REVIEW

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/instant/InstantComponentManager.kt [hunk 1]
```diff
@@ -33,7 +33,25 @@ class InstantComponentManager(
     ): InstantPaymentComponent {
         val paymentMethod = PaymentMethod.SERIALIZER.deserialize(JSONObject(encodedPaymentMethod))
         val configuration = instantPaymentConfigurationDTO.mapToCheckoutConfiguration()
-        val instantPaymentComponent = createInstantPaymentComponent(configuration, paymentMethod, componentId)
+        val instantPaymentComponent =
+            when {
+                componentId.contains(Constants.INSTANT_ADVANCED_COMPONENT_KEY) ->
```
Comment (descorp) [inline android/src/main/kotlin/com/adyen/checkout/flutter/components/instant/InstantComponentManager.kt:38]: `contains` ?
There is no way to check by ID directly?

```kotlin
when (componentId) {
   Constants.INSTANT_ADVANCED_COMPONENT_KEY -> ...
```

File: ios/Classes/components/card/advanced/CardAdvancedComponent.swift [hunk 2]
```diff
@@ -26,9 +26,10 @@ class CardAdvancedComponent: BaseCardComponent {
             componentPlatformApi: componentPlatformApi
         )
 
-        actionComponentDelegate = CardAdvancedFlowActionComponentDelegate(
+        actionComponentDelegate = ComponentActionDelegate(
             componentFlutterApi: componentFlutterApi,
-            componentId: componentId
+            componentId: componentId,
+            finalizeAndDismissComponent: finalizeAndDismiss(success:completion:)
```
Comment (descorp) [inline ios/Classes/components/card/advanced/CardAdvancedComponent.swift:32]:
```suggestion
            finalizeAndDismissHandler: finalizeAndDismiss(success:completion:)
```

File: ios/Classes/components/delegates/ComponentActionDelegate.swift [hunk 3]
```diff
@@ -1,15 +1,21 @@
 import Adyen
 
-class CardAdvancedFlowActionComponentDelegate: ActionComponentDelegate {
+class ComponentActionDelegate: ActionComponentDelegate {
```
Comment (descorp) [inline ios/Classes/components/delegates/ComponentActionDelegate.swift:3]:
```suggestion
class ComponentActionHandler: ActionComponentDelegate {
```

File: ios/Classes/components/instant/InstantAdvancedComponent.swift [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            let action = try JSONDecoder().decode(Action.self, from: jsonData)
+            actionComponent?.handle(action)
+        } catch {
+            sendErrorToFlutterLayer(error: error)
+        }
+    }
+    
+    private func onFinish(paymentEventDTO: PaymentEventDTO) {
+        let resultCode = ResultCode(rawValue: paymentEventDTO.result ?? "")
+        let success = resultCode == .authorised || resultCode == .received || resultCode == .pending
```
Comment (goergisn) [inline ios/Classes/components/instant/InstantAdvancedComponent.swift:70]: Can this be an extension on `ResultCode`?

File: ios/Classes/components/delegates/ComponentActionHandler.swift [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
+        finalizeCallback: @escaping ((Bool, @escaping (() -> Void)) -> Void)
+    ) {
         self.componentFlutterApi = componentFlutterApi
         self.componentId = componentId
+        self.finalizeCallback = finalizeCallback
     }
-
-    func didProvide(_ data: Adyen.ActionComponentData, from _: Adyen.ActionComponent) {
+    
+    internal func didProvide(_ data: Adyen.ActionComponentData, from _: Adyen.ActionComponent) {
```
Comment (goergisn) [inline ios/Classes/components/delegates/ComponentActionHandler.swift:18]:
```suggestion
    internal func didProvide(_ data: ActionComponentData, from _: ActionComponent) {
```

File: ios/Classes/components/delegates/ComponentActionHandler.swift [hunk 6]
```diff
@@ -24,24 +30,28 @@ class CardAdvancedFlowActionComponentDelegate: ActionComponentDelegate {
...
         } catch {
-            sendErrorToFlutterLayer(error: error)
+            finalizeCallback(false) { [weak self] in
+                self?.sendErrorToFlutterLayer(error: error)
+            }
         }
     }
 
-    func didComplete(from _: Adyen.ActionComponent) {
+    internal func didComplete(from _: Adyen.ActionComponent) {
```
Comment (goergisn) [inline ios/Classes/components/delegates/ComponentActionHandler.swift:39]:
```suggestion
    internal func didComplete(from _: ActionComponent) {
```

File: ios/Classes/components/delegates/ComponentActionHandler.swift [hunk 7]
```diff
@@ -24,24 +30,28 @@ class CardAdvancedFlowActionComponentDelegate: ActionComponentDelegate {
...
         // Only for voucher payment method - currently not supported.
     }
 
-    func didFail(with error: Error, from _: Adyen.ActionComponent) {
-        sendErrorToFlutterLayer(error: error)
+    internal func didFail(with error: Error, from _: Adyen.ActionComponent) {
```
Comment (goergisn) [inline ios/Classes/components/delegates/ComponentActionHandler.swift:43]:
```suggestion
    internal func didFail(with error: Error, from _: ActionComponent) {
```

**Review by goergisn — APPROVED** [review]

**Pull request merged**